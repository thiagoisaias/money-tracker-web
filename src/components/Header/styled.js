import styled from "styled-components";
import { devices } from "utils/devices";

export const Wrapper = styled.header`
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #333;
  color: #fff;

  @media ${devices.small} {
    padding: 0 16px;
  }

  @media ${devices.mediumUp} {
    padding: 0 32px;
  }
`;

export const NavBar = styled.nav`
  display: flex;
  > a {
    margin: 0 6px;
  }
`;

export const Logo = styled.div`
  color: #fff;
`;

export const Logout = styled.div`
  color: #fff;
  cursor: pointer;

  &:hover {
    opacity: 0.9;
  }
`;
