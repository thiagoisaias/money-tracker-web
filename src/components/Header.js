import React from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';


const Container = styled.header`
  height: 70px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #2d2d2d;
  color: #fff;
  border-bottom: 1px solid #e2e2e2;

  @media (max-width: 736px) {
    padding: 0 32px;
  }

  @media (min-width: 737px) {
    padding: 0 55px;
  }
`;

const Logo = styled.div`
  color: #fff;
`;

const ProfileMenu = styled.div`
  cursor: pointer;
  font-size: 24px;
`;

const Header = (props) => {
  return (
    <Container>
      <NavLink to='/'><Logo>Wallet's Logo</Logo></NavLink>
      <ProfileMenu><i className="fa fa-user" aria-hidden="true"></i></ProfileMenu>
    </Container>
  )
}

export default Header;