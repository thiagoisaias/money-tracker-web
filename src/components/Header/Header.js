import React from "react";
import { connect } from "react-redux";
import { NavLink, withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import styled from "styled-components";

import { devices } from "../../utils/devices";
import { logout } from "../../store/actions/auth/auth";

const Container = styled.header`
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: #333;
  color: #f2f2f2;

  @media ${devices.small} {
    padding: 0 16px;
  }

  @media ${devices.mediumUp} {
    padding: 0 32px;
  }
`;

const Logo = styled.div`
  color: #fff;
`;

const Logout = styled.div`
  color: #fff;
  cursor: pointer;

  &:hover {
    opacity: 0.9;
  }
`;

export const Header = props => {
  const { history, onLogout } = props;
  return (
    <Container>
      <NavLink to="/">
        <Logo>{"Money Tracker"}</Logo>
      </NavLink>
      <NavLink to="/accounts">{"Accounts"}</NavLink>
      <Logout
        onClick={() => {
          onLogout(history);
        }}
      >
        {"Logout"}
      </Logout>
    </Container>
  );
};

Header.propTypes = {
  onLogout: PropTypes.func.isRequired,
  history: PropTypes.object
};

const mapDispatchToProps = dispatch => {
  return {
    onLogout: history => {
      dispatch(logout(history));
    }
  };
};

export default withRouter(connect(null, mapDispatchToProps)(Header));
