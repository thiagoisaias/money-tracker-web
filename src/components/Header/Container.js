import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import Header from "./Header";

import { logout } from "store/actions/auth/auth";

export const Container = props => {
  const { history, logout } = props;

  return <Header history={history} logout={logout} />;
};

const mapDispatchToProps = {
  logout
};

Container.propTypes = {
  logout: PropTypes.func.isRequired,
  history: PropTypes.object.isRequired
};

export default withRouter(connect(null, mapDispatchToProps)(Container));
