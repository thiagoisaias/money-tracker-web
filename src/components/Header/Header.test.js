import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Header } from "./Header";

describe("Header", () => {
  const mockProps = {
    userName: "Lorem Foo"
  };

  const component = shallow(<Header {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
