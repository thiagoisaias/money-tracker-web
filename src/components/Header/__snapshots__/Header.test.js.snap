// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
<styled.header>
  <NavLink
    activeClassName="active"
    ariaCurrent="true"
    to="/"
  >
    <styled.div>
      Wallet's Logo
    </styled.div>
  </NavLink>
  <styled.div>
    Hi, 
    Lorem Foo
     â–¼
  </styled.div>
</styled.header>
`;
